packages/admin-ui-plugin/src/plugin.ts

```diff
@@ -278,32 +295,56 @@ export class AdminUiPlugin implements OnVendureBootstrap, OnVendureClose {
     }
 
     /**
-     * It might be that the ui-devkit compiler has not yet copied the config
-     * file to the expected location (particularly when running in watch mode),
+     * Overwrites admin-ui favicon.
+     */
+    private async overwriteAdminUiFavicon(adminUiAppPath: string, newFaviconPath?: string) {
+        if (!newFaviconPath) {
+            Logger.verbose(`Favicon not informed, keeping original.`, loggerCtx);
+            return;
+        }
+
+        const adminUiFaviconPath = path.join(adminUiAppPath, 'favicon.ico');
+        try {
+            await this.pollForFile(adminUiFaviconPath, 'favicon');
+        } catch (e) {
+            Logger.error(e.message, loggerCtx);
+            throw e;
+        }
+        try {
+            await fs.copyFileSync(newFaviconPath, adminUiFaviconPath);
+        } catch (e) {
+            throw new Error('[AdminUiPlugin] Could not write favicon.ico file:\n' + e.message);
+        }
+        Logger.verbose(`Favicon file was replaced`, loggerCtx);
+    }
+
+    /**
+     * It might be that the ui-devkit compiler has not yet copied the file
+     * to the expected location (particularly when running in watch mode),
      * so polling is used to check multiple times with a delay.
      */
-    private async pollForConfigFile(adminUiConfigPath: string) {
+    private async pollForFile(adminUiConfigPath: string, loggerFileName: string) {
         const maxRetries = 10;
         const retryDelay = 200;
         let attempts = 0;
 
         const pause = () => new Promise(resolve => setTimeout(resolve, retryDelay));
 
         while (attempts < maxRetries) {
-            try {
-                Logger.verbose(`Checking for config file: ${adminUiConfigPath}`, loggerCtx);
-                const configFileContent = await fs.readFile(adminUiConfigPath, 'utf-8');
```

Comment:
here i found out that was reading the file, but the content was not being used, so i replaced by fs.existsSync, because i think the objective of this function was try to find the file until was avaliable or pass max tries. And since i would need the same logic for waiting the favicon to be avaliable, i refactored the function to use in both operations to replace vendure config json and favicon.